lab_6_pipeline/pipeline.py

```diff
@@ -18,22 +40,52 @@ def __init__(self, path_to_raw_txt_data: Path):
         """
         Initializes CorpusManager
         """
+        self.path_to_raw_txt_data = path_to_raw_txt_data
+        self._validate_dataset()
+        self._storage = {}
+        self._scan_dataset()
 
     def _validate_dataset(self) -> None:
         """
         Validates folder with assets
         """
+        if not self.path_to_raw_txt_data.exists():
+            raise FileNotFoundError(f"No such file or directory: {self.path_to_raw_txt_data}")
+
+        if not self.path_to_raw_txt_data.is_dir():
+            raise NotADirectoryError(f"Not a directory: {self.path_to_raw_txt_data}")
+
+        meta_files = list(self.path_to_raw_txt_data.glob('*_meta.json'))
+        raw_files = list(self.path_to_raw_txt_data.glob('*_raw.txt'))
+
+        if len(meta_files) != len(raw_files):
+            raise InconsistentDatasetError('Number of meta and raw files is not equal')
+
+        raw_ind = sorted([int(file.stem.split("_")[0]) for file in raw_files])
+        for id1, id2 in zip(raw_ind, raw_ind[1:]):
+            if id2 - id1 > 1:
+                raise InconsistentDatasetError('Article IDs are not sequential')
+
+        for raw_file in raw_files:
+            if raw_file.stat().st_size == 0:
+                raise InconsistentDatasetError
+
+        if not raw_files:
```

Comment:
extremely timely check...

Reply:
fixed

Reply:
okay, I understand. This check should be done first because if the directory is empty, all subsequent checks for the existence, size, and number of files inside the directory are meaningless and redundant.